### ISSUE TITLE
ipq806x_gmac_probe() falls through to next function phy_modes()

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1240]
I've seen two 0day bot reports:
drivers/net/ethernet/stmicro/stmmac/dwmac-ipq806x.o: warning: objtool: ipq806x_gmac_probe() falls through to next function phy_modes()

on x86_64 randconfig builds.

https://lists.01.org/hyperkitty/list/kbuild-all@lists.01.org/thread/AS4WBQ24EO47AINWHVRSJVRXEFBRZ2DD/

config: https://lists.01.org/hyperkitty/list/kbuild-all@lists.01.org/message/AS4WBQ24EO47AINWHVRSJVRXEFBRZ2DD/attachment/2/config.gz

running cvise now

**Comment by nickdesaulniers** [comment]
cvise didn't converge, creduce produced:
```c
struct b {
  int (*c)(*)
};
a(char);
static phy_modes() { return ""; }
static ipq806x_gmac_probe() {
  a(phy_modes);
  __builtin_unreachable();
}
struct b d = {ipq806x_gmac_probe};
```
Maybe the `unreachable` in `ipq806x_gmac_probe` is somehow always true?

**Comment by nickdesaulniers** [comment]
```sh
$ curl -L https://lists.01.org/hyperkitty/list/kbuild-all@lists.01.org/message/AS4WBQ24EO47AINWHVRSJVRXEFBRZ2DD/attachment/2/config.gz | gunzip > .config
$ make LLVM=1 LLVM_IAS=1 -j72 olddefconfig
$ make LLVM=1 LLVM_IAS=1 -j72 drivers/net/ethernet/stmicro/stmmac/dwmac-ipq806x.o -s
```
reproduces. Seems related to:
```diff
diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac-ipq806x.c b/drivers/net/ethernet
/stmicro/stmmac/dwmac-ipq806x.c
index bf3250e0e59c..d52145788013 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-ipq806x.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-ipq806x.c
@@ -330,7 +330,7 @@ static int ipq806x_gmac_probe(struct platform_device *pdev)
                break;
        default:
                /* We don't get here; the switch above will have errored out */
-               unreachable();
+               ;
        }
        regmap_write(gmac->nss_common, NSS_COMMON_CLK_GATE, val);
```

**Comment by nickdesaulniers** [comment]
I kind of wonder why does the last switch of the 3 switches on `gmac->phy_mode` do this, but not the second?

**Comment by nickdesaulniers** [comment]
This was fixed by @nathanchance in 4367355dd90942a71641c98c40c74589c9bddf90 in 5.15.
